demisto_sdk/commands/common/update_id_set.py

```diff
@@ -136,6 +137,60 @@ def get_changed_files(files_string):
     return added_files_list, modified_files_list, added_script_list, modified_script_list
 
 
+def build_dependency_graph(playbook_data):
+    """
+    Builds tasks flow graph.
+
+    Args:
+        playbook_data (dict): playbook yml data.
+
+    Returns:
+        DiGraph: all tasks of given playbook.
+    """
+    initial_task = playbook_data.get('starttaskid', '0')
+    tasks = playbook_data.get('tasks', {})
+
+    graph = networkx.DiGraph()
+    graph.add_node(initial_task, mandatory=True)  # add starting task as root of the direct graph
+
+    found_new_tasks = True
+    while found_new_tasks:
+        current_number_of_nodes = graph.number_of_nodes()
+        leaf_nodes = {node for node in graph.nodes() if graph.out_degree(node) == 0}
+
+        for leaf in leaf_nodes:
+            leaf_task = tasks.get(leaf)
+            leaf_mandatory = graph.nodes[leaf]['mandatory']
+
+            # In this case the playbook is invalid, one of the "nexttasks" fields contain invalid task id.
+            if not leaf_task:
+                print_error(f'{playbook_data.get("id")}: No such task {leaf} in playbook')
+                continue
+                # raise ValueError(f'No such task {leaf} in playbook {playbook_data.get("id")}')
+            leaf_next_tasks = sum(leaf_task.get('nexttasks', {}).values(), [])
+
+            for task_id in leaf_next_tasks:
+                task = tasks.get(task_id)
+                if not task:
+                    print_error(f'{playbook_data.get("id")}: No such task {leaf} in playbook')
+                    continue
+                    # raise ValueError(f'No such task {leaf} in playbook {playbook_data.get("id")}')
+
+                # If task can't be skipped and predecessor task is mandatory - set as mandatory.
+                mandatory = leaf_mandatory and not task.get('skipunavailable', False)
+                if task_id not in graph.nodes():
+                    graph.add_node(task_id, mandatory=mandatory)
+                else:
+                    # If task already in graph, update mandatory field.
+                    # If one of the paths to the task is mandatory - set as mandatory.
+                    graph.nodes[task_id]['mandatory'] = graph.nodes[task_id]['mandatory'] or mandatory
```

Comment:
Should not it be just:
```
graph.nodes[task_id]['mandatory'] = mandatory
```
?

Reply:
just to be on the safe side,
if I first recognized it as mandatory and then I found another task that is optional, I don't want to override the mandatory mark. this way I can maintain the `mandatory: true` once I found it.